File: ulohy/19-tabulka-nasobku/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let cols;
let rows;

rl.on("line", (line) => {
    if (cols === undefined) {
        cols = parseInt(line);
        return;
    }

    rows = parseInt(line);

    printMultiplicationTable(rows, cols);

    rl.close();
});

function printMultiplicationTable(rows, cols) {
    let tableWidth = 1;
    for (let col = 1; col <= cols; col++) {
        tableWidth += countDigits(col * rows) + 3;
    }

    printHorizontalLine(tableWidth);

    for (let row = 1; row <= rows; row++) {
        process.stdout.write("|");

        for (let col = 1; col <= cols; col++) {
            const result = row * col;
            const columnWidth = countDigits(col * rows);
            for (let pad = 0; pad <= columnWidth - countDigits(result); pad++) {
                process.stdout.write(" ");
            }

            process.stdout.write(result + " ");
            process.stdout.write("|");
        }
        process.stdout.write("\n");

        printHorizontalLine(tableWidth);
    }
}

function countDigits(number) {
    let count = 0;
    while (number > 0) {
        number = Math.floor(number / 10);
        count++;
    }
    return count;
}

function printHorizontalLine(width) {
    for (let i = 0; i < width; i++) {
        process.stdout.write("-");
    }
    process.stdout.write("\n");
}

File: ulohy/19-tabulka-nasobku/reseni/main.py
def print_multiplication_table(rows, cols):
    table_width = 1
    for col in range(1, cols + 1):
        table_width += count_digits(col * rows) + 3

    print_horizontal_line(table_width)

    for row in range(1, rows + 1):
        print("|", end="")

        for col in range(1, cols + 1):
            result = row * col

            column_width = count_digits(col * rows)
            for pad in range(column_width - count_digits(result) + 1):
                print(" ", end="")

            print(result, end=" ")
            print("|", end="")

        print()
        print_horizontal_line(table_width)

def count_digits(number):
    count = 0
    while number > 0:
        number //= 10
        count += 1
    return count


def print_horizontal_line(width):
    for i in range(width):
        print("-", end="")
    print()


if __name__ == "__main__":
    cols = int(input())
    rows = int(input())

    print_multiplication_table(rows, cols)
